### PR TITLE
Update community-plugins.json

### PR DESCRIPTION
**Pull request by ratiger**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/ratiger/obsidian-banyan

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by Zachatoo** [comment]
[.setting-item-control {](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/styles.css#L30-L32), [.view-content .empty-state {](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/styles.css#L492-L494)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.

Please update your classes to be more specific to your plugin to avoid conflicts with other plugins.

[style={{ backgroundColor: backgroundColor ?? 'var(--background-secondary)', padding: '6px 12px', borderRadius: 8, position: "relative" }](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L83), [style={{ display: "flex", flexWrap: "wrap", overflowX: "hidden", alignItems: "center", gap: 4, rowGap: 6 }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L84), [style={{ marginLeft: 4, cursor: "pointer" }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L88), [style={{ border: "none", outline: "none", background: 'transparent' }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L100), [style={{](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L106-L116), [style={{](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/TagInput.tsx#L121-L126), [style={{ padding: '12px' }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/ConfirmModal.tsx#L52), [style={{ marginBottom: 16, fontWeight: "bold" }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/ConfirmModal.tsx#L53), [style={{ marginBottom: 16 }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/ConfirmModal.tsx#L54), [style={{ background: "var(--interactive-accent)", color: "var(--text-on-accent)", border: "none", borderRadius: 4, padding: "4px 16px" }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/ConfirmModal.tsx#L57), [style={{ display: "flex", justifyContent: "flex-end", gap: 8, marginTop: 16 }}](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/components/ConfirmModal.tsx#L55), and more
You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.

[Copyright (C) 2020-2025 by Dynalist Inc.](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/LICENSE#L1)
Please update this to the current year and your name.

["minAppVersion": "0.15.0",](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

[this.currentLanguage = moment.locale() || 'en';](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/i18n/index.ts#L17)
Please use [getLanguage](https://docs.obsidian.md/Reference/TypeScript+API/getLanguage) to get the user's current language. Note that you'll also need to update your [minAppVersion](https://docs.obsidian.md/Reference/TypeScript+API/PluginManifest/minAppVersion) to at least `1.8.0` to safely use this API.

[general_save: "save",](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/i18n/languages/en.ts#L4-L124)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

["craeted at"](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/i18n/languages/en.ts#L116)
Typo, should be `created at`

[TFile.prototype.getID = function () {](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/fileUtils.ts#L14-L16), [TFile.prototype.getTags = function (app: App) {](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/fileUtils.ts#L18-L23)
Please do not override the prototype for `TFile`, this is not safe in a plugin environment.

[async getAllFolders(): Promise<string[]> {](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/fileUtils.ts#L37-L50)
Use [`Vault.getAllFolders()`](https://docs.obsidian.md/Reference/TypeScript+API/Vault/getAllFolders) to get all folders in the vault. Note that you'll also need to update your [minAppVersion](https://docs.obsidian.md/Reference/TypeScript+API/PluginManifest/minAppVersion) to at least `1.6.6` to safely use this API.

[await this.app.vault.trash(file, true);](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/fileUtils.ts#L125)
Prefer to use [`FileManager.trashFile()`](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/trashFile) to remove a file or folder according to the user's preferred 'trash' options.

[AddCardIconEl.addClass('my-plugin-ribbon-class');](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/BanyanPlugin.ts#L34), [CardIconEl.addClass('my-plugin-ribbon-class');](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/BanyanPlugin.ts#L45), [RandomNoteIconEl.addClass('my-plugin-ribbon-class');](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/BanyanPlugin.ts#L58)
Please remove this sample plugin code.

[createFolderSuggest(](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/BanyanSettingTab.tsx#L110-L144)
Use [AbstractInputSuggest](https://docs.obsidian.md/Reference/TypeScript+API/AbstractInputSuggest) when letting a user select a file/folder to add type-ahead support.

[this.vault.on('create', this.handleCreate);](https://github.com/ratiger/obsidian-banyan/blob/779eef334e2278eeef6be21224f8ce1d9384b0ae/src/utils/fileWatcher.ts#L38-L42)
Events need to be registered, so they can be properly unloaded when the plugin is disabled. <https://docs.obsidian.md/Plugins/Events>

**Comment by joethei** [comment]
[this.app.workspace.getLeavesOfType(CARD_DASHBOARD_VIEW_TYPE).forEach(leaf => leaf.detach());](https://github.com/ratiger/obsidian-banyan/blob/07c8e894fe7b2bb21dd38dbd782062a76262defb/src/BanyanPlugin.ts#L66)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[random_review_title_create: "Create Random Review",](https://github.com/ratiger/obsidian-banyan/blob/07c8e894fe7b2bb21dd38dbd782062a76262defb/src/utils/i18n/languages/en.ts#L116-L117)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const hasFileID = (content: string) => {](https://github.com/ratiger/obsidian-banyan/blob/07c8e894fe7b2bb21dd38dbd782062a76262defb/src/models/FileInfo.ts#L25-L44)
Use the `processFrontMatter` function instead.

**Comment by ratiger** [comment]
@joethei hi, I've updated my `manifest.json` file in my repo. Is it OK from there or am I supposed to do something more?
